### PR TITLE
Add client error sending functionality in client/bidirectional streaming scenarios

### PR DESCRIPTION
**Pull request by dilanSachi**
## Purpose
The error is sent to the server using an `RSTFrame` which resets the stream. The server receives a error with message `Remote client closed the connection while reading inbound request entity body`.
## Checklist
- [x] Linked to an issue
- [ ] Updated the changelog
- [x] Added tests

Fixes https://github.com/ballerina-platform/ballerina-standard-library/issues/386

### CODE REVIEW

File: ballerina-tests/tests/60_client_send_error_in_client_bidi_streaming.proto [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+// specific language governing permissions and limitations
+// under the License.
+
+syntax = "proto3";
+
+import "google/protobuf/wrappers.proto";
+
+service ErrorSendService {
+	rpc sendErrorClientStreaming(stream google.protobuf.StringValue) returns (google.protobuf.Int32Value);
+  rpc sendErrorBidiStreaming(stream google.protobuf.StringValue) returns (stream google.protobuf.Int32Value);
```
Comment (BuddhiWathsala) [inline ballerina-tests/tests/60_client_send_error_in_client_bidi_streaming.proto:23]: Indentation?

File: ballerina-tests/tests/60_client_send_error_in_client_bidi_streaming_client.bal [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+import ballerina/grpc;
+import ballerina/test;
+import ballerina/lang.runtime;
+
+@test:Config {enable:true}
+isolated function testClientStreamingSendErrorToService() returns grpc:Error? {
+    ErrorSendServiceClient errorClient = check new ("http://localhost:9160");
+    SendErrorClientStreamingStreamingClient streamingClient = check errorClient->sendErrorClientStreaming();
+
+    check streamingClient->sendString("Hey");
```
Comment (BuddhiWathsala) [inline ballerina-tests/tests/60_client_send_error_in_client_bidi_streaming_client.bal:26]: Here, I don't understand the intention behind these tests. You are sending an error first and then create a new client and send correct messages again. Don't we need to reuse the same streaming client here?

Comment (dilanSachi) [reply ballerina-tests/tests/60_client_send_error_in_client_bidi_streaming_client.bal:26]: Im trying to check whether the service received an error in the previous client connection

File: ballerina-tests/tests/60_client_send_error_in_client_bidi_streaming_service.bal [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+}
+service "ErrorSendService" on ep60 {
+    remote function sendErrorClientStreaming(stream<string, error?> clientStream) returns int {
+        error? e = clientStream.forEach(isolated function(string val) {
+            io:println(val);
+        });
+        if e is () {
+            return clientStreamingErrorCount;
+        } else {
+            io:println("Received a client error : " + e.message());
```
Comment (daneshk) [inline ballerina-tests/tests/60_client_send_error_in_client_bidi_streaming_service.bal:37]: shall we log API to print the message? Normally we don't `io:println` in server-side implementation

File: ballerina-tests/tests/60_client_send_error_in_client_bidi_streaming_client.bal (hunk 2 above)
Comment (BuddhiWathsala) [reply ballerina-tests/tests/60_client_send_error_in_client_bidi_streaming_client.bal:26]: Okay, got it.

**Review by BuddhiWathsala — APPROVED** [review]
LGTM

File: ballerina-tests/tests/60_client_send_error_in_client_bidi_streaming.proto (hunk 1 above)
Comment (dilanSachi) [reply ballerina-tests/tests/60_client_send_error_in_client_bidi_streaming.proto]: Changed in https://github.com/ballerina-platform/module-ballerina-grpc/pull/444/commits/379da88fe9706c7fd054608e8b041974a6951308

File: ballerina-tests/tests/60_client_send_error_in_client_bidi_streaming_service.bal (hunk 3 above)
Comment (dilanSachi) [reply ballerina-tests/tests/60_client_send_error_in_client_bidi_streaming_service.bal]: Changed in https://github.com/ballerina-platform/module-ballerina-grpc/pull/444/commits/f0cc1d136eb72b71d1a97a917281c0b1fa9cb9a1

**Review by daneshk — APPROVED** [review]
LGTM

**Review by BuddhiWathsala — APPROVED** [review]

**Pull request merged**